tools/ds-identify

```diff
@@ -1565,6 +1565,25 @@ dscheck_VMware() {
     return "${DS_NOT_FOUND}"
 }
 
+dscheck_WSL() {
+    if [ ! "${DI_VIRT}" = "wsl" ]; then
+        return "${DS_NOT_FOUND}"
+    fi
+
+    # The datasource needs to find the cloud-config files in the Windows host
+    # filesystem, which is exposed as 9p mount points, one per disk drive (partition).
+    # If none is found, the datasource cannot proceed.
+    # See https://youtu.be/lwhMThePdIo?t=2431&si=JKTHx39TyRgPbzkZ and
+    # https://learn.microsoft.com/en-us/windows/wsl/wsl-config#what-is-drvfs
+    # for more information.
+    if [ "$(grep -c '^\S* \S* 9p \S*aname=drvfs;.*' "${PATH_ROOT}/proc/mounts")" = 0 ]; then
```

Comment:
I don't think that `\S` is a [Posix regex](https://en.wikibooks.org/wiki/Regular_Expressions/POSIX_Basic_Regular_Expressions), which we prefer for portability.

From Gnu grep's `info` page I see:

```info
‘\S’
     Match non-whitespace, it is a synonym for ‘[^[:space:]]’.
```

So I think we want somthing more like:
```suggestion
    if [ "$(grep -c '^[^[:space:]]* [^[:space:]]* 9p [^[:space:]]*aname=drvfs;.*' "${PATH_ROOT}/proc/mo  unts")" = 0 ]; then
```

Which also works in Gnu and busybox (cc: @dermotbradley) grep implementations:
```bash
$ echo "C:\\134 /mnt/c 9p rw,dirsync,aname=drvfs;path=C:\\;uid=0;" | grep -c '^[^[:space:]]* [^[:space:]]* 9p [^[:space:]]*aname=drvfs;.*'
1
$ echo "C:\\134 /mnt/c 9p rw,dirsync,aname=drvfs;path=C:\\;uid=0;" | busybox grep -c '^[^[:space:]]* [^[:space:]]* 9p [^[:space:]]*aname=drvfs;.*'
1
```